returnn/hdf.py

```diff
@@ -281,3 +281,71 @@ def run(self):
             audio.close()
 
         out_hdf.close()
+
+
+class RasrAlignmentDumpHDFJob(Job):
+    """
+    This Job reads Rasr alignment caches and dump them in hdf files.
+    """
+
+    def __init__(
+        self,
+        alignment_caches: List[tk.Path],
+        allophone_file: tk.Path,
+        state_tying_file: tk.Path,
+        returnn_root: Optional[tk.Path] = None,
+    ):
+        """
+        :param list[Path]: alignment_caches
+        :param Path: allophone_file
+        :param Path: state_tying_file
+        :param Optional[Path] returnn_root: file path to the RETURNN repository root folder
+        """
+        self.alignment_caches = alignment_caches
+        self.allophone_file = allophone_file
+        self.state_tying_file = state_tying_file
+        self.out_hdf_files = [self.output_path(f"data.hdf.{d}", cached=False) for d in range(len(alignment_caches))]
+        self.returnn_root = returnn_root
+        self.rqmt = {"cpu": 1, "mem": 8, "time": 0.5}
+
+    def tasks(self):
+        yield Task("run", resume="run", rqmt=self.rqmt, args=range(1, (len(self.alignment_caches) + 1)))
+
+    def run(self, task_id):
+        from i6_core.lib.rasr_cache import FileArchive
+        import numpy as np
+
+        state_tying = dict(
+            (k, int(v)) for l in open(self.state_tying_file.get_path()) for k, v in [l.strip().split()[0:2]]
+        )
+
+        alignment_cache = FileArchive(
+            self.alignment_caches[min(task_id - 1, len(self.alignment_caches) - 1)].get_path()
```

Comment:
why the minimum here? I think `task_id - 1` should always be correct. And even if not, then we dump the same file twice to different outputs which is probably not expected, so maybe rather fail with a KeyError here.

Reply:
Oh yes, this was from another logic, where it was possible to have less alignment caches than hdf files. But here we do not need this.